src/test/java/seedu/address/model/attendance/HoursTest.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.attendance;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class HoursTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Hours(null));
+    }
+
+    @Test
+    public void constructor_invalidHours_throwsIllegalArgumentException() {
+        String invalidHours = "10 hours";
+        assertThrows(IllegalArgumentException.class, () -> new Hours(invalidHours));
+    }
+
+    @Test
+    void isValidHours() {
+
+        // invalid hours
+        assertFalse(Hours.isValidHours(null)); // null
+        assertFalse(Hours.isValidHours("")); // empty string
+        assertFalse(Hours.isValidHours(" ")); // spaces only
+        assertFalse(Hours.isValidHours("twenty")); // wrong format
+        assertFalse(Hours.isValidHours("2hrs")); // wrong format
+        assertFalse(Hours.isValidHours("3.5")); // wrong format
```

Comment:
I think we can add negative values here, should we add in maximum and negative values of int -  2147483647 and -2147483647, this may be overkill though but we can check for overflow then?

Reply:
Yup I'll add in testing of overflow. Do you think we should include that in our message constraint? (E.g. "Hours should be a non-negative integer up to 2147483647 and it should not be blank")

Reply:
Hmm maybe we don't add it in the message constrain but the UG? I think it won't look really nice if we display a large number, but let's get the others' opinion on this.